### ISSUE TITLE
multi-input/output LSTM regression

### ISSUE THREAD

**Issue by ch779** [keras-team/keras#7639]
Hey guys, I'm new to this area, just got my hands on it for a couple of weeks.

Problem: I need to build a network to model a dynamic System with multi-inputs and outputs. 
so I decide to predict a single value for each of my outputs at time t+1 based on multiple Inputs through the past, say 10 time-steps.

I take here 2 Inputs and 3 Outputs. Every Input and Output is a vector (y_values-time). I intended to build for both Inputs a separate LSTM layer, and the two Layers come together and then give 3 outputs.

questions: 0. any suggestions what kind of net architecture and  type might work? 
                1. should I merge the two Inputs (same size) into one Matrix and feed it into LSTM?
                2.  should I prepare my whole Input vector into fractions (batches) and feed once in the network or should I build a Loop to cut my Input(s) into sections for each Iteration?
               3. how should I define my batch_shape in functional API
 
here is the code, I can't get it running and I'm a little confused about the shape of the data, can you enlighten me on this? many thx.

#Setup
import ...

#Prepare data
t= pd.read_table...
lw= pd.read_table...
gas= pd.read_table...
psi= pd.read_table...
accx= pd.read_table...
accy= pd.read_table...

#define data structure and reshape data
batch_start= 0
batch_size= 10
input_length= 1
target_length= 1

t= t.values
lw= lw.values
gas= gas.values
psi= psi.values
accx= accx.values
accy= accy.values

#Build network
input1= Input(batch_shape= (batch_size, input_length), name='input1')
input2= Input(batch_shape= (batch_size, input_length), name='input2')

L1= LSTM(10, stateful= True, return_sequences= True)(input1)
L2= LSTM(10, stateful= True, return_sequences= True)(input2)

target= concatenate([L1, L2])
target1= TimeDistributed(Dense(target_length, activation= 'linear'))(target)
target2= TimeDistributed(Dense(target_length, activation= 'linear'))(target)
target3= TimeDistributed(Dense(target_length, activation= 'linear'))(target)

model= Model(
    inputs= [input1, input2],
    outputs= [target1, target2, target3]
    )

model.compile(optimizer= 'rmsprop', loss= 'mean_squared_error')
plot_model(model, to_file= 'my_net.png', show_shapes= True, show_layer_names= True)

#Train network
#for step in range() ... build a loop

t_fit= t[batch_start:batch_size*input_length]
lw_fit= lw[batch_start:batch_size*input_length]
gas_fit= gas[batch_start:batch_size*input_length]
psi_fit= psi[batch_start:batch_size*input_length]
accx_fit= accx[batch_start:batch_size*input_length]
accy_fit= accy[batch_start:batch_size*input_length]

#update batch_start
batch_start += batch_size*input_length 

model.fit(
    [lw_fit, gas_fit],
    [psi_fit, accx_fit, accy_fit]
    )

**Comment by samre12** [comment]
@ch779 just to clarify, are you inputs one dimensional for each timestep, i.e. each input that you have, is it `[timesteps, ]` dimensional?

**Comment by ch779** [comment]
@samre12  thx for your reply.
it's correct, both of my inputs are, when I take it as a time sequence, a vector, so 2 dimensional tensor, right?
at every time, the value is just a number. 
my system outputs can be determined by the history of the inputs. does it not mean that I have to use memoryable net (rnn, lstm)?
can you write a line for me to show me how to feed my input( shape) into a convolutional network?
so I have to prepare my Inputdata to be sth like (1, timesteps) for each iteration?

**Comment by samre12** [comment]
@ch779, if your horizon, as in the number of timesteps that you look backward (and it should be a small number) to make a decision is fixed, then a convolutional net is as effective than using a LSTM.
However, you could take a look at the paper on [DeepSense](https://arxiv.org/abs/1611.01942).

**Comment by samre12** [comment]
@ch779 please close this issue if you think it is resolved. :)

**Comment by ch779** [comment]
I currently built a Conv1D NN to predict my three outputs, its a regression problem.

so I merge my 3 Input data into a (timesteps, 3) input matrix and I iterate my window of timesteps, say 10, through my whole Input matrix and my target data is the 3 outputs at the next time step.
I trained the net through out the whole Input and I want to check if it can give the right output as in the training set, but my model.predict is a straight line. can you check below if sth is wrong? (I can do live prediction, put my model.predict in the loop, it works fine, but I need the final model so I can feed in new Inputs) 

so sth like:

## I/O dimension, shape of data
batch_start= 0
time_steps= 10
input_length= 3
target_length= 3
## build model
model= Sequential()
model.add(Conv1D(64, 6, input_shape= (time_steps, input_length)))
model.add(Conv1D(32, 5))
model.add(Flatten())
model.add(Dense(64))
model.add(Dropout(0.25))
model.add(Dense(64))
model.add(Dense(32))
model.add(Dense(16))
model.add(Dense(8))
model.add(Dense(3))
model.compile(optimizer= 'rmsprop', loss= 'mean_squared_error')
## train model
for i in range(len(t)-batch_size*time_steps):
    t_fit= t[batch_start:batch_start+time_steps] 
    input_ts_fit= input_ts[batch_start:batch_start+time_steps] ## input matrix in timesteps window
    target_ts_fit= target_ts[batch_start+time_steps] ## the output at t+1

    input_ts_fit= input_ts_fit[np.newaxis,:,:] ## fit the dimension of data
    target_ts_fit= target_ts_fit[np.newaxis,:]

    batch_start += 1
    ## fit model
    model.fit(input_ts_fit, target_ts_fit, batch_size=5, epochs= 10)
## predict data based on training example
batch_start= 0
for i in range(len(t)-batch_size*time_steps):
    ## data preparation as the same above
    pred_ts = model.predict(input_ts_fit)
    batch_start += 1
    ## plot the result
